src/controllers/controllermanager.h

```diff
@@ -58,6 +63,8 @@ class ControllerManager : public QObject {
     void requestShutdown();
     void requestSave(bool onlyActive);
     void requestInitialize();
+    void keyboardPresetChanged(KeyboardControllerPresetPointer pKbdPreset);
+    void keyboardEnabled(bool enabled);
 
   public slots:
     void updateControllerList();
```

Comment:
It does more or less what it is supposed to.

 1. First, it calls its enumerators to query their devices. `KeyboardController` is instantiated directly since it has no enumerator and is added to [the new list](https://github.com/mixxxdj/mixxx/pull/966/files#diff-ca4c4a4a94ebc7ecf996e143fe8cee5fR180), with its _Controller cousins_, if you like.
 2. If the new list is different than the current one, replace the old one. Hence the name; update the controller list.

However, I see one problem. Imagine the following scenario where **DJ Tomasito** * has one HID controller hooked up. [`m_controllers`](https://github.com/mixxxdj/mixxx/blob/master/src/controllers/controllermanager.h#L98) would look like this:
 1. `KeyboardController A`
 2. `HidController A`

However, when **DJ Tomasito** decides to connect a second HID controller and updates the controller list with an imaginary _Refresh Controller List_ button in controller preferences, `m_controllers` should look like this:
 1. `KeyboardController A`
 2. `HidController A`
 3. `HidController B`

Right?

Well. It turns out that not all `queryDevices()` implementations clear their own [`m_devices`](https://github.com/mixxxdj/mixxx/blob/master/src/controllers/hid/hidenumerator.h#L21) list before returning it. [They just append to it](https://github.com/mixxxdj/mixxx/blob/master/src/controllers/hid/hidenumerator.cpp#L102), without clearing the current list. So, in this case, **DJ Tomasito**'s controller list would look like this:

 1. `KeyboardController A`
 2. `HidController A`
 3. `HidController A`  _← Not good_
 4. `HidController B`

There's just one enumerator which actually does clear its list (and deletes its devices) before querying new devices, [`PortMidiEnumerator`](https://github.com/Tomasito665/mixxx/blob/develop/src/controllers/midi/portmidienumerator.cpp#L185). I think that if the other enumerators do the same, that this `updateControllerList()` method will work as it is supposed to.

However, I think this is outside of the scope of this PR. We might do this in a future PR: _Update Controller List from Controller Preferences_. If we do this, we would probably use this method as a slot, connecting it to a button in the GUI, so I think we can let it as it is, for now.

Also, I lack the controllers to actually test this right now.

---
\* _DJ Tomasito_ is fictional, I am just a jazz pianist :)